test/Identity.IntegrationTest/Endpoints/IdentityServerTests.cs

```diff
@@ -179,6 +178,255 @@ public async Task TokenEndpoint_GrantTypePassword_WrongAuthEmailHeader_Fails()
+            { "deviceType", DeviceTypeAsString(DeviceType.FirefoxBrowser) },
+            { "deviceIdentifier", deviceId },
+            { "deviceName", "firefox" },
+            { "grant_type", "password" },
+            { "username", username },
+            { "password", "master_password_hash" },
+        }), context => context.SetAuthEmail(username));
+
+        Assert.Equal(StatusCodes.Status400BadRequest, context.Response.StatusCode);
+
+        var body = await AssertHelper.AssertResponseTypeIs<JsonDocument>(context);
+        var root = body.RootElement;
+
+        var error = AssertHelper.AssertJsonProperty(root, "error", JsonValueKind.String).GetString();
+        Assert.Equal("invalid_grant", error);
+        var errorDescription = AssertHelper.AssertJsonProperty(root, "error_description", JsonValueKind.String).GetString();
+        Assert.StartsWith("sso authentication", errorDescription.ToLowerInvariant());
+    }
+
+    [Theory]
+    [BitAutoData(OrganizationUserType.Owner)]
+    [BitAutoData(OrganizationUserType.Admin)]
+    public async Task TokenEndpoint_GrantTypePassword_WithOwnerOrAdmin_WithSsoPolicyEnabled_WithEnforceSsoPolicyForAllUsersFalse_Success(OrganizationUserType organizationUserType, int generatedUsername)
+    {
+        var organizationId = new Guid("5e77c2cd-f84b-4a27-affa-12c08d1adbda");
+        var deviceId = "92b9d953-b9b6-4eaf-9d3e-11d57144dfeb";
+        var username = $"{generatedUsername}@example.com";
+
+        var server = _factory.WithWebHostBuilder(builder =>
+        {
+            builder.UseSetting("globalSettings:sso:enforceSsoPolicyForAllUsers", "false");
+        }).Server;
+
+        await server.PostAsync("/accounts/register", JsonContent.Create(new RegisterRequestModel
+        {
+            Email = username,
+            MasterPasswordHash = "master_password_hash"
+        }));
+
+        await CreateOrganizationWithSsoPolicy(organizationId, username, organizationUserType, ssoPolicyEnabled: true);
+
+        var context = await server.PostAsync("/connect/token", new FormUrlEncodedContent(new Dictionary<string, string>
+        {
+            { "scope", "api offline_access" },
+            { "client_id", "web" },
+            { "deviceType", DeviceTypeAsString(DeviceType.FirefoxBrowser) },
+            { "deviceIdentifier", deviceId },
+            { "deviceName", "firefox" },
+            { "grant_type", "password" },
+            { "username", username },
+            { "password", "master_password_hash" },
+        }), context => context.SetAuthEmail(username));
+
+        Assert.Equal(StatusCodes.Status200OK, context.Response.StatusCode);
+    }
+
+    [Theory]
+    [BitAutoData(OrganizationUserType.Owner)]
+    [BitAutoData(OrganizationUserType.Admin)]
+    public async Task TokenEndpoint_GrantTypePassword_WithOwnerOrAdmin_WithSsoPolicyDisabled_WithEnforceSsoPolicyForAllUsersTrue_Success(OrganizationUserType organizationUserType, int generatedUsername)
```

Comment:
I think you could combine this with the next test, everyone should receive a success result in this instance. (Although the next test has `EnforceSsoPolicyForAllUsersFalse`, I'm not sure why that's different?) 

Or if you want to keep both tests to cover all combinations of the policy + global setting (which is also fine), you should add non-owner/admins to this test, and add owners/admins to the next test. They should still pass.